### PR TITLE
Fix proxy-based nayduck tests so that they can run on non-unix systems.

### PR DESCRIPTION
**Pull request by robin-near**
Before this, running proxy-based nayduck tests (such as proxy_simple.py) fails on Mac because on Mac, multiprocessing.Process uses spawn, not fork, and our tests were written in a way that was unfriendly to spawn:

1. the entry point was not protected by `if __name__ == '__main__':`, causing spawned processes to re-execute the main module's code;
2. shared memory was not properly passed to the child process - we relied on referencing the same global variable which only worked with the fork implementation.

This PR fixes these. Also, re-enable two tests which are now fixed.

### CODE REVIEW

File: pytest/tests/sanity/network_drop_package.py [hunk 1]
```diff
@@ -51,16 +47,20 @@ async def handle(self, msg, fr, to):
         return not drop
 
 
-start_cluster(3, 0, 1, None, [["epoch_length", 500]], {}, Handler)
+if __name__ == '__main__':
+    success = Value('i', 0)
+
+    start_cluster(3, 0, 1, None, [["epoch_length", 500]], {},
+                  partial(Handler, success=success))
```
Comment (wacban) [inline pytest/tests/sanity/network_drop_package.py:54]: personally I prefer multiline arguments, if you like that too you can achive that with default formatter by adding trailing comma in the argument list

File: pytest/tests/sanity/proxy_example.py [hunk 2]
```diff
@@ -20,12 +20,13 @@
...
 TIMEOUT = 30
-success = Value('i', 0)
 
 
 class Handler(ProxyHandler):
 
-    def __init__(self, *args, **kwargs):
+    def __init__(self, *args, success=None, **kwargs):
+        assert success is not None
+        self.success = success
```
Comment (wacban) [inline pytest/tests/sanity/proxy_example.py:29]: I see this logic repeated, could it go to ProxyHandler?

File: pytest/tests/sanity/sync_ban.py [hunk 3]
```diff
@@ -19,97 +19,112 @@
...
         return True
 
 
-node0_config = {
-    "consensus": {
-        "min_block_production_delay": {
-            "secs": 0,
-            "nanos": 400000000
+if __name__ == '__main__':
+    should_ban = sys.argv[1] == 'true'
```
Comment (wacban) [inline pytest/tests/sanity/sync_ban.py:63]: not in this PR but we should use a proper library for parsing args (argparse is ok)

File: pytest/tests/sanity/sync_ban.py [hunk 4]
```diff
@@ -19,97 +19,112 @@
...
+    node0_config = {
+        "consensus": {
+            "min_block_production_delay": {
+                "secs": 0,
+                "nanos": 1000000000
```
Comment (wacban) [inline pytest/tests/sanity/sync_ban.py:68]: why this change?

File: pytest/tests/sanity/sync_ban.py [hunk 5]
```diff
@@ -19,97 +19,112 @@
...
+                    status1['sync_info']['syncing'] is False):
+                break
+        time.sleep(2)
+
+    if not should_ban and (tracker0.check(BAN_STRING) or
+                           tracker1.check(BAN_STRING)):
+        assert False, "unexpected ban of peers"
+
+    # logger.info('shutting down')
+    # time.sleep(10)
```
Comment (wacban) [inline pytest/tests/sanity/sync_ban.py:130]: What about this?

**Review by wacban — APPROVED** [review]
Sweet! 

<a href="https://imgflip.com/i/7snkbw"></a><div></div>

File: pytest/tests/sanity/proxy_example.py (hunk 2 above)
Comment (robin-near) [reply pytest/tests/sanity/proxy_example.py:29]: Some of them don't use the flag. The main issue however is that if it goes into the ProxyHandler there's no easy way for the parent process (main code) to access it.

File: pytest/tests/sanity/sync_ban.py (hunk 3 above)
Comment (robin-near) [reply pytest/tests/sanity/sync_ban.py:63]: Yeah... I would have also wished the tests not even to be in python :(

File: pytest/tests/sanity/sync_ban.py (hunk 4 above)
Comment (robin-near) [reply pytest/tests/sanity/sync_ban.py:68]: The original test wanted it produce blocks too fast that the syncer chouldn't catch up as quickly (idk why but I don't wanna debug this moment). So I slowed it down.

File: pytest/tests/sanity/sync_ban.py (hunk 5 above)
Comment (robin-near) [reply pytest/tests/sanity/sync_ban.py:130]: It isn't needed. I should've removed the two lines though, I'll do that next time :)

Note that this test was not even enabled before this PR.
